src/main/java/com/webapp/TextBook/validation/SharedValidation/loginUserInfoValidation/LoginUserInfoValidatorImpl.java

```diff
@@ -0,0 +1,125 @@
+import javax.validation.ConstraintValidatorContext;
+import java.util.ArrayList;
+
+import com.webapp.TextBook.validation.Shared.RegexPatternContainer;
+
+import com.webapp.TextBook.validation.Shared.ErrorBinding;
+import com.webapp.TextBook.validation.Shared.ErrorBindingException;
+import com.webapp.TextBook.validation.Shared.SharedValidationState;
+import com.webapp.TextBook.viewModel.sharedViewModel.loginUserInfo.LoginUserInfo;
+
+public class LoginUserInfoValidatorImpl implements
+        ConstraintValidator<LogInUserInfoValidationInterface, LoginUserInfo>,
+        SharedValidationState {
+
+
+    private boolean haveSuffix;
+
+    @Override
+    public void initialize(LogInUserInfoValidationInterface info){
+            this.haveSuffix = info.haveSuffix();
+
+    }
+
+    @Override
+    public boolean isValid(LoginUserInfo user, ConstraintValidatorContext constraintContext){
+
+
+        ArrayList<ErrorBinding<String>> errorList = new ArrayList<ErrorBinding<String>>();
+
+        // holds if any inputs are null
+        boolean userInputsNull = false;
+
+        // initial check -- any fields are null (faulty form data)
+        if(user.get_password() == null){
+            userInputsNull = true;
+            errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_PASSWORD, "Password field is empty", null));
+        }
+        else if(user.get_username() == null){
+            userInputsNull = true;
+            errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_USERNAME, "Username field is empty", null));
+        }
+
+
+        try {
+
+            if(userInputsNull){
+
+                //if binding error list is not empty then set constraint validator’s message
+                // to the abstract method’s json creator helper.
+                constraintContext.disableDefaultConstraintViolation();
+
+                constraintContext.buildConstraintViolationWithTemplate(
+                        ErrorBinding.ErrorBindingJsonHelper.CreateJsonStringFromErrorBindings(errorList));
+
+
+                return false;
+            }
+
+
+            // commence bi-if logic -- failure of prerequisite condition does not
+            // bar second input (password) from being validated
+            if (!user.get_username().matches(RegexPatternContainer.STUDENT_ID_PATTERN.pattern())) {
+                // add to error binding list
+                errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_USERNAME, "Invalid 919 number. Please follow format (919######).", user.get_username()));
+            }
+
+            // second input validation (password)
+            // tests prefix of password
+            // two variations exist: password w/ or w/o domain
+            if (!user.get_password().matches(RegexPatternContainer.S_NUMBER_PREFIX.pattern())) {
+                errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_PASSWORD, "Password does not follow S-Number form (S######). Please revise", null));
```

Comment:
1. This is a nitpick but you say format for 919 and form for snumber. Should be consistent 
2. What isn't nitpicky is that we shouldn't be conceding the information that the pw is the s number. This could be a security threat since sNumbers are easily accessible.

Reply:
Request to mark resolved: this is validation (your portion)